### DISCUSSION TITLE
Design needed: Display selection settings

### DISCUSSION THREAD

**Discussion by AtomCrafty** [ppy/osu#12998] [Ideas]
What we need is a design for the display selection dialog. The idea is to show a representation of the user's monitor setup (a bunch of boxes arranged as reported by the OS). The user should then be able to simply click one of them to select the corresponding display as the new output device.

Some ideas we developed on discord:
It might be appropriate to move it to some sort of popup menu to avoid cluttering the main settings menu too much. We've also had the suggestion to access said menu with a button next to the regular screen dropdown (similar to the "Browse" button next to file selection prompts).
If there is only one display connected the option shouldn't show up at all (like the resolution dropdown).

![Mockup](https://user-images.githubusercontent.com/4553050/46289024-0252b900-c588-11e8-8ddc-b5afd22b30ac.PNG)

**Comment by smoogipoo** [comment]
Do we really need this, @peppy ?

**Comment by peppy** [comment]
As mentioned in discord during this discussion, I was envisaging a similar control to be used for letterbox sizing/positioning. This could potentially be part of that control.

It also paves the way for non-fullscreen borderless display (which is something that has been requested in the past), as there is a method of moving the window without having a title bar displayed.

My concerns with adding this functionality is cross-platform compatibility.
